tools/component-builder/css/vars.js

```diff
@@ -44,26 +44,39 @@ function bakeVars() {
     // Find all custom properties used in the component
     let variableList = varUtils.getVarsFromCSS(file.contents);
 
+    // Get vars defined inside of the component
+    let componentVars = varUtils.getVarsDefinedInCSS(file.contents);
+
     // Get vars in ALL components
     let vars = await varUtils.getAllComponentVars();
 
     // Get literally all of the possible vars (even overridden vars that are different between themes/scales)
     let allVars = await varUtils.getAllVars();
 
     // For each color stop and scale, filter the variables for those matching the component
+    let errors = [];
     let usedVars = {};
     variableList.forEach(varName => {
       if (varName.indexOf('spectrum-global') !== -1) {
-        logger.warn(`${pkg.name} directly uses global variable ${varName}`);
+        logger.warn(`⚠️  ${pkg.name} directly uses global variable ${varName}`);
       }
       else if (!allVars[varName]) {
-        logger.warn(`${pkg.name} uses undefined variable ${varName}`);
+        if (componentVars.indexOf(varName) === -1) {
+          errors.push(`${pkg.name} uses undefined variable ${varName}`);
+        }
+        else {
+          logger.warn(`🔶 ${pkg.name} uses locally defined variable ${varName}`);
```

Comment:
This is new: previously, locally defined variables were issued the same warning (`uses undefined variable`). Now they get their own warning, and it's not a failure.